can-validate/map/validate/validate.js

```diff
@@ -325,7 +326,7 @@ proto.__set = function (prop, value, current, success, error) {
 		// If validate opts are set and initing, validate properties only if validateOnInit is true
 		if ((validateOpts && !mapIniting) || (validateOpts && mapIniting && validateOpts.validateOnInit)) {
 			// Validate item
-			allowSet = this._validateOne({key: prop, value: value}, validateOpts);
+			allowSet = this._validateOne({key: prop, value: value}, validateOpts, this.attr());
```

Comment:
Just tested and it looks like `.attr()` doesn't retrieve the computed value from the getter (although it maybe should). For this to work, we'd need to look inside validateOpts, extract the property we need and `.attr(prop)` at this point. Not sure that's a great solution, though.